src/Collections/WorkflowLogsCollection.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+namespace Pantheon\Terminus\Collections;
+
+use GuzzleHttp\Exception\GuzzleException;
+use Pantheon\Terminus\Exceptions\TerminusException;
+use Pantheon\Terminus\Models\Environment;
+use Pantheon\Terminus\Models\Site;
+use Pantheon\Terminus\Models\TerminusModel;
+use Pantheon\Terminus\Models\WorkflowLog;
+
+/**
+ * Class WorkflowLogsCollection
+ * @package Pantheon\Terminus\Collections
+ */
+class WorkflowLogsCollection extends SiteOwnedCollection implements \Iterator
+{
+    /**
+     *
+     */
+    public const PRETTY_NAME = 'Workflow Logs Collection';
+    /**
+     * @var string
+     */
+    protected $collected_class = WorkflowLog::class;
+    /**
+     * @var string
+     */
+    protected $url = 'sites/{site_id}/logs/workflows';
+    /**
+     * @var
+     */
+    private $current;
+
+    /**
+     * @throws TerminusException
+     */
+    public function __construct(Site $site)
+    {
+        parent::__construct(["site" => $site]);
+    }
+
+    /**
+     * @throws TerminusException
+     * @throws GuzzleException
+     */
+    public function fetch()
+    {
+        $result = $this->request()->request(
+            $this->getUrl(),
+            ['site_id' => $this->getSite()->id, "method" => "GET"]
+        );
+        $this->setData($result['data']);
+        return $this;
+    }
+
+    /**
+     * @param array $data
+     * @return void
+     */
+    public function setData(array $data = [])
+    {
+        $class = $this->getCollectedClass();
+        foreach ($data as $datum) {
+            $this->models[] = new $class($datum, ['collection' => $this]);
+        }
+    }
+
+
+    /**
+     * @throws TerminusException
+     * @throws GuzzleException
+     */
+    public function serialize()
+    {
+        return array_map(function ($model) {
+            return $model->serialize();
+        }, $this->models);
+    }
+
+    /**
+     * @return mixed|\Pantheon\Terminus\Models\TerminusModel
+     */
+    public function current(): ?TerminusModel
+    {
+        return $this->models[$this->current];
+    }
+
+    /**
+     * @return void
+     */
+    public function next(): void
+    {
+        $this->current++;
+    }
+
+    /**
+     * @return mixed|null
+     */
+    public function key(): int
+    {
+        return $this->current;
+    }
+
+    /**
+     * @return bool
+     */
+    public function valid(): bool
+    {
+        return isset($this->models[$this->current]);
+    }
+
+    /**
+     * @return void
+     */
+    public function rewind(): void
+    {
+        $this->current = 0;
+    }
+
+    public function findLatestFromOptionsArray(
+        $options = [
+            'type' => null,
+            'id' => null,
+            'commit_hash' => null,
+            'start' => 0,
+        ]
+    ): TerminusModel {
+        $start_time = $options['start'] ?? 0; // Default to 0 if not set
+        $wfl = $this->latest();
+
+        // If the workflow is newer than the start time and matches the options
+        if (
```

Comment:
This `if` block is not correct; I think the easiest fix would be to delete it.  It does not make sense to search for a workflow by ID **and** commit_hash. It also doesn't really make sense to search by workflow ID **and** workflow type, since there is only one workflow with a given ID. Does it ever make sense to search for a workflow by commit hash and type? I think perhaps not, since workflows with commit hashes should always be sync_code, or some variant thereof. If I am wrong about this (e.g. if upstream_update and friends also have commit_hash in their workflow fields), then perhaps searching by type needs to be pushed down into findLatestByProperty, just as $start_time was.